pkg/api/binding.go

```diff
@@ -0,0 +1,63 @@
+package api
+
+import (
+	bindingApi "github.com/redhat-developer/service-binding-operator/apis/binding/v1alpha1"
+	specApi "github.com/redhat-developer/service-binding-operator/apis/spec/v1alpha3"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// ServiceBinding describes a service binding, from group binding.operators.coreos.com/v1alpha1 or servicebinding.io/v1alpha3
+type ServiceBinding struct {
+	Name   string                `json:"name"`
+	Spec   ServiceBindingSpec    `json:"spec"`
+	Status *ServiceBindingStatus `json:"status,omitempty"`
+}
+
+type ServiceBindingSpec struct {
+	Services               []specApi.ServiceBindingServiceReference `json:"services"`
+	DetectBindingResources bool                                     `json:"detectBindingResources"`
+	BindAsFiles            bool                                     `json:"bindAsFiles"`
+}
+
+type ServiceBindingStatus struct {
+	BindingFiles   []string `json:"bindingsFiles,omitempty"`
+	BindingEnvVars []string `json:"bindingEnvVars,omitempty"`
+}
+
+// ServiceBindingFromBinding returns a common api.ServiceBinding structure
+// from a ServiceBinding.binding.operators.coreos.com/v1alpha1
+func ServiceBindingFromBinding(binding bindingApi.ServiceBinding) (ServiceBinding, error) {
```

Comment:
Do we really need the `error` return type in the signature? We are always returning `nil` as `error` here. Removing it would simplify callers' code by removing an error handling branch.